Impossible-Agent-Suite/SwapFees/src/pairs.fetcher.ts

```diff
@@ -0,0 +1,27 @@
+import { providers, Contract, BigNumber } from "ethers";
+import abi from "./abi";
+
+export default class PairFetcher {
+  readonly factory: string;
+  private fContract: Contract;
+
+  constructor(factory: string, provider: providers.Provider) {
+    this.factory = factory;
+    this.fContract = new Contract(factory, abi.FACTORY, provider);
+  }
+
+  public async getAllPairs(block: number | string): Promise<string[]> {
+    const length: number = await this.fContract.allPairsLength({ blockTag: block });
+
+    const pairPromises: Promise<string>[] = [];
+    for (let i = BigNumber.from(0); i.lt(length); i = i.add(1)) {
+      pairPromises.push(
+        this.fContract
+          .allPairs(i, { blockTag: block })
```

Comment:
In a similar fashion to the [flash loan agent](https://github.com/NethermindEth/Forta-Agents/pull/145) there may be memory issues if we get all pairs from the factory and keep them stored in memory. If you think this could be a potential issue then an implementation similar to what is linked in the flash loan agent (LRU cache and query for unknown addresses)